examples/v1beta1/general-purpose.yaml

```diff
@@ -11,6 +11,9 @@ spec:
     expireAfter: Never
   template:
     spec:
+      startupTaints:
+        - key: node.cilium.io/agent-not-ready
```

Comment:
```suggestion
        # If you are using cilium dataplane you might notice that there is some overprovisioning where we launch extra nodes to schedule pods that remain empty and are later removed. 
        # this taint will prevent that scenario, read the docs here: https://karpenter.sh/docs/faq/#why-do-i-see-extra-nodes-get-launched-to-schedule-pending-pods-that-remain-empty-and-are-later-removed
        - key: node.cilium.io/agent-not-ready
```

Reply:
Maybe some context here to make it clear this is just if you are using cilium. network unavailable during cni initialization should also have this same side effect.